imap_processing/cdf/cdf_attribute_manager.py

```diff
@@ -76,10 +76,42 @@ def __init__(self, data_dir: Path):
         self.variable_attribute_schema = self._load_default_variable_attr_schema()
 
         # Load Default IMAP Global Attributes
-        self.global_attributes = CdfAttributeManager._load_yaml_data(
+        self._global_attributes = CdfAttributeManager._load_yaml_data(
             self.source_dir / DEFAULT_GLOBAL_CDF_ATTRS_FILE
         )
-        self.variable_attributes = dict()
+        self._variable_attributes = dict()
+
+    @property
+    def global_attributes(self):
+        """Moving global_attributes to property."""
+        # Validating required global attributes are present
+        for attr_name in self.global_attribute_schema.keys():
+            required_attr = self.global_attribute_schema[attr_name]["required"]
+            if required_attr is True and attr_name not in self._global_attributes:
+                self._global_attributes[attr_name] = None
+
+        return self._global_attributes
+
+    @global_attributes.setter
+    def global_attributes(self, instrument_id: str):
```

Comment:
This setter has me confused. I think of setters being used to modify some data in the class, but I don't see that anything is being modified here. The only possible side-effect of calling `self.get_global_attributes(instrument_id)` in this setter is that it validates the global attributes for the input instrument by raising an exception if a required global variable is not defined.

Reply:
Thank you! This is my first time working with properties, so I will appreciate all the feedback I can get. I updated the property itself and eliminated the setter, so it makes much more sense now.